### DISCUSSION TITLE
[question] how to map a filetype to another parser language?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [nvim-treesitter/nvim-treesitter#3628] [General]
Does there exist any method to map a filetype to another filetype in order to use its fold/indent/highlight? Like coc.nvim's `g:coc_filetype_map`. Thanks!

```
g:coc_filetype_map					*g:coc_filetype_map*

	Map for document filetypes so the server could handle current document
	as another filetype, example: >

	let g:coc_filetype_map = {
		\ 'html.swig': 'html',
		\ 'wxss': 'css',
		\ }
<
	Default: {}
```

**Comment by theHamsta** [comment]
From readme:

If you wish to set a specific parser for a filetype, you should extend the filetype_to_parsername table:
```lua
local ft_to_parser = require"nvim-treesitter.parsers".filetype_to_parsername
ft_to_parser.someft = "python" -- the someft filetype will use the python parser and queries.

 ```

**Reply by Freed-Wu** [reply]
For some filetype, e.g., liquid, it is a template language to generate html/css/json/etc. [tpope/vim-liquid](https://github.com/tpope/vim-liquid) will set `b:liquid_subtype` to `html`/`css`/`json`/etc. How to set it dynamically? Thanks.

Similar problem: <https://github.com/neoclide/coc.nvim/issues/4282>

**Reply by rafaelpirolla** [reply]
2024 update, just if someone search engine drops here. Form README.md:
If you wish to set a specific parser for a filetype, you should use vim.treesitter.language.register():
```
vim.treesitter.language.register('python', 'someft')  -- the someft filetype will use the python parser and queries.
```
Note this requires Nvim v0.9.
